core/src/main/java/org/apache/lucene/analysis/miscellaneous/DuplicateByteSequenceSpotter.java

```diff
@@ -0,0 +1,325 @@
+        // one-off paths
+        ByteSequenceLeafNode[] children = null;
+
+        LightweightTreeNode(byte key, TreeNode parentNode, int depth) {
+            super(key, parentNode, depth);
+            bytesAllocated += LIGHTWEIGHT_TREE_NODE_OBJECT_SIZE;
+
+        }
+
+        @Override
+        public short add(int byteSequence) {
+            if (children == null) {
+                // Create array adding new child
+                children = new ByteSequenceLeafNode[1];
+                bytesAllocated += RamUsageEstimator.NUM_BYTES_ARRAY_HEADER;
+
+                ByteSequenceLeafNode child = new ByteSequenceLeafNode(byteSequence);
+                bytesAllocated += RamUsageEstimator.NUM_BYTES_OBJECT_REF;
+                children[0] = child;
+                return child.hitCount;
+            }
+            // Find existing child and increment count
+            for (ByteSequenceLeafNode child : children) {
+                if (child.byteSequence == byteSequence) {
+                    if (child.hitCount < Short.MAX_VALUE) {
+                        child.hitCount++;
+                    }
+                    return child.hitCount;
+                }
+            }
+            // Grow array adding new child
+            ByteSequenceLeafNode[] newChildren = new ByteSequenceLeafNode[children.length + 1];
+            bytesAllocated += RamUsageEstimator.NUM_BYTES_OBJECT_REF;
+
+            System.arraycopy(children, 0, newChildren, 0, children.length);
+            children = newChildren;
+            ByteSequenceLeafNode child = new ByteSequenceLeafNode(byteSequence);
+            children[newChildren.length - 1] = child;
+            nodesResizedByDepth++;
+            return child.hitCount;
+        }
+
+        @Override
+        public TreeNode add(byte b, int depth) {
+            throw new UnsupportedOperationException("Leaf nodes do not take byte sequences");
+        }
+
+    }
+
+    // Leaf node implementation that represents a sequence of bytes as an int
+    // for a compact representation
+    class ByteSequenceLeafNode {
+        short hitCount;
+        int byteSequence;
+
+        ByteSequenceLeafNode(int byteSequence) {
+            bytesAllocated += LEAF_NODE_OBJECT_SIZE;
+            this.byteSequence = byteSequence;
+            nodesAllocatedByDepth[3]++;
+            hitCount = 1;
+        }
+    }
+
+    public final long getEstimatedSizeInBytes() {
+        return bytesAllocated;
+    }
+
+    /**
+     * @return Performance info - the number of nodes allocated at each depth
+     */
+    public int[] getNodesAllocatedByDepth() {
+        return nodesAllocatedByDepth.clone();
+    }
+
+    /**
+     * @return Performance info - the number of resizing of children arrays, at
+     *         each depth
+     */
+    public int getNodesResizedByDepth() {
+        return nodesResizedByDepth;
+    }
+
+    public int getDuplicateSequenceSize() {
```

Comment:
Might be legacy. Previously I had variable-depth trie size and allocated TreeNode objects for each level/branch. This method was a reminder of the tree depth passed in the constructor.

Later I opted for a more space-efficient impl with no option for user-supplied depths.  It is fixed at 6 which seems to work well and this may be useful for clients to know but there's already a static constant.
I'll bin it.